tools/config_header_gen.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2022, seL4 Project a Series of LF Projects, LLC
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+import argparse
+import sys
+import yaml
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Generate a configuration C header from a configuration YAML file.",
+    )
+    parser.add_argument(
+        "in_file",
+        nargs="?",
+        type=argparse.FileType("r"),
+        default=sys.stdin,
+        help="Input YAML file.",
+    )
+    parser.add_argument(
+        "-o",
+        dest="out_file",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+        help="Output C header file.",
```

Comment:
What about a `--lang=<c|.....>¸ parameter?

Reply:
Would that be to leave room for growth, or do you have specific alternative languages or formats in mind?

Reply:
Well, both actually. I have nothing specific really in mind, this would just come handy for some experimenting. There are some places already where tools implicitly assume it's C (or ARM), and I think it's good practice to make such choices explicit.

Reply:
Not sure that makes sense here -- the yaml file is supposed to be the interface for other languages, and this tool is the one that re-establishes the seL4 C headers from that. If we're adding support for other languages, it's not clear that we'd want to change this tool. We'd usually instead provide a separate tool that parses yaml (as we're doing for Rust), and it would not be in the kernel cmake build.

We could rename it slightly to make explicit that it is for C, but the help text already does say that.

Reply:
Point taken. I'd argue that it might be a chance to enforce unification of tools. But it's hard to argue given the python script can already be used amost as a library. Maybe just rework it, so the `main()` calls a function with parameters then?

Reply:
The change mentioned in https://github.com/seL4/seL4/pull/975#issuecomment-1449088299 has resulted in a structure for the script that matches what you are describing, @axel-h.